card/mod/carrierwave/set/type/image.rb

```diff
@@ -9,46 +9,60 @@
     _render_core size: :icon
   end
 
-  view :source, cache: :never do
-    determine_image_size
-    source_url
-  end
-
-  def source_url
+  view :source do
     return card.raw_content if card.web?
-    internal_url selected_version.url
+    image = selected_version
+    # return "" unless image.valid?
+    internal_url image.url
   end
-
+  
   def selected_version
-    if voo.size == :original
-      card.image
+    size = determine_image_size
+    if size && size != :original
+      card.image.versions[size]
     else
-      card.image.versions[voo.size.to_sym]
+      card.image
     end
   end
 
-  def default_core_args _args={}
-    determine_image_size
+  def handle_source
+    super
+  end
+
+  def closed_size
+    :icon
+  end
+
+  def main_size
+    :large
+  end
+
+  def default_size
+    :medium
   end
 
   def determine_image_size
     voo.size =
       case
-      when nest_mode == :closed then :icon
+      when nest_mode == :closed then closed_size
       when voo.size             then voo.size.to_sym
-      when main?                then :large
-      else                           :medium
+      when main?                then main_size
+      else                           default_size
```

Comment:
this is a newer change and means different formats can have different default sizes.  Potentially useful..